extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/datasource/ReactiveStreamCommandsImpl.java

```diff
@@ -370,12 +370,14 @@ protected XPendingSummary decodeAsXPendingSummary(Response r) {
         }
 
         var pending = r.get(0).toLong();
-        var lowest = r.get(1).toString();
-        var highest = r.get(2).toString();
+        var lowest = r.get(1) != null ? r.get(1).toString() : null;
+        var highest = r.get(2) != null ? r.get(2).toString() : null;
 
         Map<String, Long> consumers = new HashMap<>();
-        for (Response nested : r.get(3)) {
-            consumers.put(nested.get(0).toString(), nested.get(1).toLong());
+        if (r.get(3) != null) {
+            for (Response nested : r.get(3)) {
+                consumers.put(nested.get(0).toString(), nested.get(1).toLong());
```

Comment:
In there we are sure the values are not null? If so, LGTM!

Reply:
I'm pretty sure that both the keys and the values are never `null`, but not completely, to be honest.

Reply:
me neither - it's not really said on the command documentation. I think it's good to go.